### PR TITLE
New API's

### PR DESCRIPTION
**Pull request by sonu0610**
### Added new APIs in Checkout Service

**GeneratePayment**
   - This API is created for generating 500 error.
   - **Endpoint:** /payment/send

**GenerateSalesTax**
   - This API is created for generating 408 error if country provided in query parameter is 'france'.
   - **Endpoint:** /cart/calculateTax

**GenerateCartEmpty**
   - This API is created for delaying the response from API as per delay mentioned in query parameter. (Default delay will be 5 seconds)
   - **Endpoint:** /cart/clear

### CODE REVIEW

File: src/frontend/handlers.go [hunk 1]
```diff
@@ -304,6 +304,88 @@ func (fe *frontendServer) viewCartHandler(w http.ResponseWriter, r *http.Request
...
+
+	// GRPC client API call of checkoutService
+	// API request  is formed with fetched data
+	_, err := pb.NewCheckoutServiceClient(fe.checkoutSvcConn).
+		GenerateSalesTax(r.Context(), &pb.GenerateSalesTaxRequest{
+			Country: country,
+		})
+
+	// Handle error
+	// Return error with 408 status code.
```
[the remark this answers is not in the document]
Comment (sonu0610) [reply src/frontend/handlers.go:360]: Done

File: src/checkoutservice/main.go [hunk 2]
```diff
@@ -168,6 +168,58 @@ func (cs *checkoutService) Watch(req *healthpb.HealthCheckRequest, ws healthpb.H
...
+	// doSomeProcessing() is called for some processing delay in API call
+	doSomeProcessing()
+
+	var country = "france"
+
+	// if country data provided in request is france then return error
+	// else return success
+	if req.Country == country {
+		time.Sleep(2 * time.Second)
+		if ctx.Err() == context.Canceled {
```
Comment (dmitryax) [inline src/checkoutservice/main.go:193]: It should be a select statement here like this:

```
		select {
		case <-ctx.Done():
			return fmt.Errorf("request is cancelled or timed out %w", err)
		case <-time.After(3 * time.Second):
		}
```

File: src/checkoutservice/main.go [hunk 3]
```diff
@@ -168,6 +168,58 @@ func (cs *checkoutService) Watch(req *healthpb.HealthCheckRequest, ws healthpb.H
...
+			return nil, status.Error(codes.Canceled, "Cancelled the request")
+		}
+		return new(pb.Empty), status.Errorf(codes.Internal, "Something went wrong with this request!")
```
Comment (dmitryax) [inline src/checkoutservice/main.go:196]: this line can be removed since it cannot be reached

Comment (sonu0610) [reply src/checkoutservice/main.go]: Done

File: src/checkoutservice/main.go (hunk 2 above)
Comment (sonu0610) [reply src/checkoutservice/main.go]: Done

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Please rebase

**Comment by sonu0610** [comment]
Done with Rebase. Please check and let me know for any updates.

**Comment by dmitryax** [comment]
May be something wrong with the github. Force merge worked out

**Pull request merged**